jetty-openid/src/test/java/org/eclipse/jetty/security/openid/CredentialsDecoderTest.java

```diff
@@ -0,0 +1,91 @@
+//
+//  ========================================================================
+//  Copyright (c) 1995-2019 Mort Bay Consulting Pty. Ltd.
+//  ------------------------------------------------------------------------
+//  All rights reserved. This program and the accompanying materials
+//  are made available under the terms of the Eclipse Public License v1.0
+//  and Apache License v2.0 which accompanies this distribution.
+//
+//      The Eclipse Public License is available at
+//      http://www.eclipse.org/legal/epl-v10.html
+//
+//      The Apache License v2.0 is available at
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class CredentialsDecoderTest
+{
+    public static Stream<Arguments> paddingExamples()
+    {
+        return Stream.of(
+            Arguments.of("XXXX", "XXXX"),
+            Arguments.of("XXX", "XXX="),
+            Arguments.of("XX", "XX==")
```

Comment:
would add `XXX=` and maybe `X=` to see what happens

Reply:
How do you think it should behave with input `X=`. Right now it would still pad that to `X===`, but this actually an illegal base64 string length without the padding.

Maybe we should not pad any string which ends with `=`, or perhaps we should just throw illegal argument because it is already padded.

Reply:
I could see a poorly implemented OP send back padded base64-encoded JWT sections. In this case, I think we should just try to use it as the OP gave and not add padding. Or, if the OP didn't pad enough, maybe we should add the required amount of padding. I'm just trying to make sure that we're as tolerant of what we receive while being as strict about what it is we send.

Reply:
Okay, I changed `padJWTSection()` so we no longer try to add padding if the string ended with `=`.